opencensus/trace/propagation/jaeger_format.py

```diff
@@ -0,0 +1,121 @@
+        The value of enabled parsed from header is int. Need to convert to
+        bool.
+
+        :type header: str
+        :param header: Trace context header which was extracted from the HTTP
+                       request headers.
+
+        :rtype: :class:`~opencensus.trace.span_context.SpanContext`
+        :returns: SpanContext generated from the trace context header.
+        """
+        if header is None:
+            return SpanContext()
+
+        try:
+            match = re.search(_UBER_HEADER_RE, header)
+        except TypeError:
+            logging.warning(
+                'Header should be str, got {}. Cannot parse the header.'
+                .format(header.__class__.__name__))
+            raise
+
+        if match:
+            trace_id = match.group(1)
+            span_id = match.group(2)
+            trace_options = match.group(4)
+
+            if trace_options is None:
+                trace_options = 1
+
+            span_context = SpanContext(
+                trace_id=trace_id,
+                span_id=span_id,
+                trace_options=TraceOptions(str(int('0x' + trace_options,16))),
+                from_header=True)
+            return span_context
+        else:
+            logging.warning(
+                'Cannot parse the header {}, generate a new context instead.'
+                .format(header))
+            return SpanContext()
+
+    def from_headers(self, headers):
+        """Generate a SpanContext object using the trace context header.
+
+        :type headers: dict
+        :param headers: HTTP request headers.
+
+        :rtype: :class:`~opencensus.trace.span_context.SpanContext`
+        :returns: SpanContext generated from the trace context header.
+        """
+        if headers is None:
+            return SpanContext()
+        if _UBER_HEADER_NAME not in headers:
+            return SpanContext()
+        return self.from_header(headers[_UBER_HEADER_NAME])
+
+    def to_header(self, span_context):
+        """Convert a SpanContext object to header string.
+
+        :type span_context:
+            :class:`~opencensus.trace.span_context.SpanContext`
+        :param span_context: SpanContext object.
+
+        :rtype: str
+        :returns: A trace context header string in google cloud format.
+        """
+        trace_id = span_context.trace_id
+        span_id = span_context.span_id
+        trace_options = span_context.trace_options.trace_options_byte
+
+        header = '{}:{}::{:02x}'.format(
+            trace_id,
+            span_id,
+            int(trace_options))
+        return header
+
+    def to_headers(self, span_context):
+        """Convert a SpanContext object to HTTP request headers.
+
+        :type span_context:
+            :class:`~opencensus.trace.span_context.SpanContext`
+        :param span_context: SpanContext object.
+
+        :rtype: dict
+        :returns: Trace context headers in google cloud format.
```

Comment:
and one more

Reply:
@SergeyKanzhelev you'd think I'd know how to use sed or something! Sorry about that, fixed and pushed.